content/en/docs/reference/networking/virtual-ips.md

```diff
@@ -0,0 +1,337 @@
+
+In these proxy models, the traffic bound for the Service's IP:Port is
+proxied to an appropriate backend without the clients knowing anything
+about Kubernetes or Services or Pods.
+
+If you want to make sure that connections from a particular client
+are passed to the same Pod each time, you can select the session affinity based
+on the client's IP addresses by setting `.spec.sessionAffinity` to `ClientIP`
+for a Service (the default is `None`).
+
+### Session stickiness timeout
+
+You can also set the maximum session sticky time by setting
+`.spec.sessionAffinityConfig.clientIP.timeoutSeconds` appropriately for a Service.
+(the default value is 10800, which works out to be 3 hours).
+
+{{< note >}}
+On Windows, setting the maximum session sticky time for Services is not supported.
+{{< /note >}}
+
+## IP address assignment to Services
+
+Unlike Pod IP addresses, which actually route to a fixed destination,
+Service IPs are not actually answered by a single host.  Instead, kube-proxy
+uses packet processing logic (such as Linux iptables) to define _virtual_ IP
+addresses which are transparently redirected as needed.
+
+When clients connect to the VIP, their traffic is automatically transported to an
+appropriate endpoint. The environment variables and DNS for Services are actually
+populated in terms of the Service's virtual IP address (and port).
+
+### Avoiding collisions
+
+One of the primary philosophies of Kubernetes is that you should not be
+exposed to situations that could cause your actions to fail through no fault
+of your own. For the design of the Service resource, this means not making
+you choose your own port number if that choice might collide with
+someone else's choice.  That is an isolation failure.
+
+In order to allow you to choose a port number for your Services, we must
+ensure that no two Services can collide. Kubernetes does that by allocating each
+Service its own IP address from within the `service-cluster-ip-range`
+CIDR range that is configured for the API server.
+
+To ensure each Service receives a unique IP, an internal allocator atomically
+updates a global allocation map in {{< glossary_tooltip term_id="etcd" >}}
+prior to creating each Service. The map object must exist in the registry for
+Services to get IP address assignments, otherwise creations will
+fail with a message indicating an IP address could not be allocated.
+
+In the control plane, a background controller is responsible for creating that
+map (needed to support migrating from older versions of Kubernetes that used
+in-memory locking). Kubernetes also uses controllers to check for invalid
+assignments (e.g. due to administrator intervention) and for cleaning up allocated
+IP addresses that are no longer used by any Services.
+
+#### IP address ranges for Service virtual IP addresses {#service-ip-static-sub-range}
+
+{{< feature-state for_k8s_version="v1.25" state="beta" >}}
+
+Kubernetes divides the `ClusterIP` range into two bands, based on
```

Comment:
This paragraph is migrated and rewritten?

Reply:
Yes.